### PR TITLE
Add Roblox v0.539.0.5390403

### PR DESCRIPTION
**Pull request by aaronliu0130**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free. 
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues). It appears that the refused ones were due to the submitter not correctly using an `installer` stanza and closed to stale.
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --new-cask <cask>` worked successfully. It turns out the installer is signed but the app itself isn't signed.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by bevanjkay** [comment]
@aaronliu0130 Unfortunately the script installer is interactive, so this would need to use  [`installer manual`](https://docs.brew.sh/Cask-Cookbook#installer-manual) unless you can find a way to suppress the GUI.

A maintainer or another contributor will be able to help with the livecheck.

**Comment by bevanjkay** [comment]
@aaronliu0130 Unfortunately, the installer is interactive so this cask would need to use [`installer manual`](https://docs.brew.sh/Cask-Cookbook#installer-manual) unless you can find a way to suppress the GUI. 

A maintainer or another contributor will be able to help with the livecheck.

**Comment by p-linnane** [comment]
I'm looking into this. We can get the latest URL from the location header in "https://www.roblox.com/download/client". I haven't been able to find anything that references the half of the version before the comma. Their release notes aren't helpful.

@bevanjkay Are we ok with just using what can be pulled from those headers for the version number?

**Comment by aaronliu0130** [comment]
>> Also, the history of roblox.rb shows that it was removed and attempted to be readded a few times since. Not sure if we're opening a can of worms here.

> It appears that the refused ones were due to the submitter not correctly using an installer stanza and closed to stale.

After a lot of searching for a silent mode, I went ahead and changed the installer to manual.

I also realized that the installer gives you a dock icon i.e. pins the roblox icon to dock. Which file should I delete to remove this icon?

**Comment by aaronliu0130** [comment]
>> We can get the latest URL from the location header in "https://www.roblox.com/download/client".

After looking at the delete pr, it seems like the download string can be parsed from https://clientsettingscdn.roblox.com/v1/client-version/MacPlayer, in the key `clientVersionUpload`, after the `version-`. However, the version string appears to be incorrect.

**Comment by aaronliu0130** [comment]
I think the awaiting user reply label can be removed now.

**Comment by aaronliu0130** [comment]
Is the part of the version before the comma important? The installer never seems to change for each version bump, and the app will auto-update itself. I also couldn't find a page that changes with it.

File: Casks/roblox.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+cask "roblox" do
+  version "1.2.2.5360459,6e3d2e7c823142ae"
+  sha256 "eefc95fb85c888cbefaa16212779954c46482197442a82555f2e622135a28f06"
+
+  url "https://setup.rbxcdn.com/mac/version-#{version.csv.second}-Roblox.dmg",
+      verified: "setup.rbxcdn.com/"
+  name "Roblox"
+  desc "Online multiplayer game platform"
+  homepage "https://www.roblox.com/"
+
```
Comment (bevanjkay) [inline Casks/roblox.rb:10]: Pending https://github.com/Homebrew/brew/pull/13658 - here is a livecheck that will retrieve everything that we need.

```suggestion
  livecheck do
    url "https://clientsettingscdn.roblox.com/v1/client-version/MacPlayer"
    regex(/version[._-]([^"]+)/i)
    strategy :page_match do |page, regex|
      client_version = page[regex, 1]
      next if client_version.blank?

      cask = CaskLoader.load("roblox")
      download_url = "https://setup.rbxcdn.com/mac/version-#{client_version}-Roblox.dmg"
      main_version = Homebrew::Livecheck::Strategy::ExtractPlist.find_versions(cask: cask, url: download_url)[:matches].values.first
      next if main_version.blank?

      "#{main_version},#{client_version}"
    end
  end
```

Comment (aaronliu0130) [reply Casks/roblox.rb:10]: Thanks! I will merge it all into one commit after the ci passes.

Comment (aaronliu0130) [reply Casks/roblox.rb:10]: Thanks! Do I still have to merge them into one commit?

Comment (aaronliu0130) [reply Casks/roblox.rb:10]: I'll probably merge the commits after the relevant pull is merged.

**Comment by aaronliu0130** [comment]
Something really strange just happened. Roblox updated and now the version is the version we see at the appcast url
image
{"version":"0.538.0.5380364","clientVersionUpload":"version-17f793e683ec4fd2","bootstrapperVersion":"5380364"}

File: Casks/roblox.rb (hunk 1 above)
Comment (aaronliu0130) [reply Casks/roblox.rb:10]: I just updated the version and added a currently working livecheck, is this good?

**Comment by aaronliu0130** [comment]
It has been 15 days, I don't see what's blocking this pull other than squashing

**Comment by bevanjkay** [comment]
@aaronliu0130 I am seeing the latest version`1.2.0.5390403` in app.

```
	<key>CFBundleShortVersionString</key>
	<string>1.2.0.5390403</string>
```

**Comment by aaronliu0130** [comment]
image
It's this version for me.

**Comment by bevanjkay** [comment]
@aaronliu0130 Because the download is an installer, not the actual application, and we need to use `installer manual`, the installer is downloaded to the caskroom, and then must be run manually by the user.

**Comment by aaronliu0130** [comment]
But the file in the screenshot IS the app itself

**Comment by bevanjkay** [comment]
@aaronliu0130 Roblox is a bit tricky because we are only downloading the installer, not the actual application. I've had a look at trying to download the actual application but I'm not sure that it is working correctly. I will push the changes, and can you give it a test?

**Comment by bevanjkay** [comment]
@aaronliu0130 The issue has been traced and will hopefully be fixed soon

**Comment by aaronliu0130** [comment]
Wait, actually there's a Roblox.app inside the downloaded app's MacOS folder which IS signed.
image

**Comment by aaronliu0130** [comment]
Nevermind, it was the installer.

**Comment by aaronliu0130** [comment]
This probably isn't something a contributor can fix without reverting to the "run the installer app" way. The problem is that the upstream devs didn't bother to sign the actual app, they only signed the installer.

**Comment by miccal** [comment]
Thank you for your submission @aaronliu0130, but we cannot accept it at this time because it does not pass our signature verification check:

> macOS on ARM requires applications to be signed. Please contact the upstream developer to let them know they should sign their app.

Please consider hosting this [in your own Tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap) and if the developers sign their app in the future we would then be happy to consider adding this to `homebrew-cask`.

**Pull request closed without merging**